src/sphinx_substitution_extensions/shared.py

```diff
@@ -0,0 +1,14 @@
+from typing import Dict
+
+from docutils.parsers.rst import directives  # type: ignore
+from docutils.parsers.rst import Directive
+from sphinx.directives.code import CodeBlock
+
+_EXISTING_DIRS = directives._directives  # pylint: disable=protected-access
```

Comment:
These variables are imported from elsewhere but really should not be as they are marked with underscores.
Instead - change the filename to start with an underscore and leave these without underscores.